packages/engine.test/src/testProducer.ts

```diff
@@ -0,0 +1,104 @@
+expect.extend({
+    toMatchExpectations(mockCalls, verbExpectations) {
+        // const remainingExpectations = verbExpectations.slice(0)
+        // const remainingMockCalls = mockCalls.slice(0)
+    
+        const isLonger = mockCalls.length !== verbExpectations.length
+
+        if(isLonger) return {
+            message: () => (`expected calls ${this.utils.printReceived(mockCalls)} to be same length as expectation ${this.utils.printExpected(verbExpectations)}`),
+            pass: false
+        }
+
+        let isSame: boolean = true
+        for(let i = 0; i < mockCalls.length; i++) {
+            expect(mockCalls[i][0]).toEqual(verbExpectations[i])
+
+            // if(verbExpectations[i].match) {
+            //     expect(mockCalls[i][0]).toMatchObject(verbExpectations[i].match)
+            // }
+            // if(verbExpectations[i].equal) {
+            //     expect(mockCalls[i][0]).toEqual(verbExpectations[i].equal)
+            // }
+        }
+
+        // const pass = true
+      
+  
+        // if (pass) {
+        //     return {
+        //         message: () => (`expected ${this.utils.printReceived(received)} not to contain object ${this.utils.printExpected(argument)}`),
+        //         pass: true
+        //     }
+        // } else {
+        //     return {
+        //         message: () => (`expected ${this.utils.printReceived(received)} to contain object ${this.utils.printExpected(argument)}`),
+        //         pass: false
+        //     }
+        // }
+    }
+  })
+
+jest.useFakeTimers();
+
+export function testProducer(name: string, {
+    producer, 
+    presets = {}, 
```

Comment:
Rename `prests` to `values`

Reply:
they aren't really values because you can also add mocked modules

Reply:
The mocked modules and other values are the values of the properties of the object that the producer receives as it's the first argument.

From the point of view of the producer, those are the input values. That's why I think we should be clear about what it represents.

Also, a mock should be indistinguishable from a real value from the producer's perspective and concerns.

Reply:
yeah but do you consider a function to be a value?

Reply:
hmm... why wouldn't it be a value? the producer receives an object with values, those values can be primitive or non-primitive, but from the point of view of the producer, they are values.